frontends/web/src/routes/account/walletconnect/index.tsx

```diff
@@ -0,0 +1,474 @@
+      const { request } = params;
+      let message: string;
+      const activeSessions = Object.values(web3wallet?.getActiveSessions());
+      const currentSession = activeSessions.find((session) => session.topic === topic);
+
+      switch (request.method) {
+      case EIP155_SIGNING_METHODS.ETH_SIGN:
+        /**
+        * Wallet Connect's ETH_SIGN gives the params as [address, message]
+        * while PERSONAL_SIGN gives them as [message, address]
+        */
+        message = request.params[1];
+        if (currentSession) {
+          handleSessionRequest(async () => {
+            const result = await ethSignMessage(code, message);
+            if (result.success) {
+              const response = { id, jsonrpc: '2.0', result: result.signature };
+              return { response, success: true };
+            }
+
+            return { success: false, error: result };
+          },
+          currentSession, topic, id, 'Sign Message', message);
+        }
+        return;
+      case EIP155_SIGNING_METHODS.PERSONAL_SIGN:
+        message = request.params[0];
+        if (currentSession) {
+          handleSessionRequest(async () => {
+            const result = await ethSignMessage(code, message);
+            if (result.success) {
+              const response = { id, jsonrpc: '2.0', result: result.signature };
+              return { response, success: true };
+            }
+
+            return { success: false, error: result };
+          },
+          currentSession, topic, id, 'Sign Message', message);
+        }
+        return;
+      case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA:
+      case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V3:
+      case EIP155_SIGNING_METHODS.ETH_SIGN_TYPED_DATA_V4:
+        const typedData = JSON.parse(request.params[1]);
+        if (currentSession) {
+          handleSessionRequest(async () => {
+            // If the typed data to be signed includes its own chainId, we use that, otherwise use the id in the params
+            const chainId = typedData.domain.chainId ?
+              +typedData.domain.chainId :
+              +params.chainId.split(':')[1];
+            const result = await ethSignTypedMessage(code, chainId, request.params[1]);
+            if (result.success) {
+              const response = { id, jsonrpc: '2.0', result: result.signature };
+              return { response, success: true };
+            }
+
+            return { success: false, error: result };
+          },
+          currentSession, topic, id, 'Sign Typed Data', JSON.stringify(typedData));
+        }
+        return;
+      //TODO: for the following two cases, let user set own gas price and nonce in BBApp
+      case EIP155_SIGNING_METHODS.ETH_SEND_TRANSACTION:
+        if (currentSession) {
+          handleSessionRequest(async () => {
+            const result = await ethSignWalletConnectTx(code, true, +params.chainId.split(':')[1], request.params[0]);
```

Comment:
that `+` business looks like JS hackery :joy: consider the explicit `parseInt()`.

what if chainID does not contain a `:`?

maybe `params.chainID.replace(/^eip155\:/, '')` is better to strip of the prefix, as then it's also clear what the format is at the same time.

Reply:
That's no hackery :D
Standard Unary Plus: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Unary_plus
imo completely sufficient for this usecase :) 
https://i.stack.imgur.com/LLrgj.png

chainID will always be in the 'eip155:' format, but I will replace it with the regex 👍